### PR TITLE
Fix behavior of CROSS JOIN on parenthesized table expressions

### PR DESCRIPTION
**Pull request by Sairakan**
When APPLY was originally implemented, a subtle issue was introduced
whereby queries that invoked CROSS JOIN with a parenthesized table expression
on the right side would raise a syntax error. This commit reverts that change
and instead implements APPLY via an ANTLR rewrite.
 
### Issues Resolved

BABEL-2805

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/input/BABEL-2805.sql [hunk 1]
```diff
@@ -33,3 +33,135 @@ GO
...
+)
+SELECT temp_table_3.[Audit.ColumnName],
+temp_table_3.[Audit.OldValue],
+temp_table_3.[Audit.NewValue]
+FROM temp_table_3
+WHERE temp_table_3.[Audit.Id] = @p1
+AND temp_table_3.[Audit.InsertOnly] = @p2;
+GO
+
+-- Regression tests for CROSS JOIN
```
Comment (2jungkook) [inline test/JDBC/input/BABEL-2805.sql]: Please add the below tests to the upgrade test files. Specifically, we need tests using views and procedures.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -986,6 +986,34 @@ class tsqlCommonMutator : public TSqlParserBaseListener
...
+            else if (ctx->OUTER())
+            {
+                /* replace OUTER with LEFT */
+                rewritten_query_fragment.emplace(std::make_pair(ctx->OUTER()->getSymbol()->getStartIndex(),
+                                                                std::make_pair(ctx->OUTER()->getText(), "LEFT")));
+                rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
+                                                                std::make_pair(ctx->APPLY()->getText(), "JOIN LATERAL")));
+                /* Need to add an always true join qualifier in order to satisfy PG grammar */
+                rewritten_query_fragment.emplace(std::make_pair(ctx->getStop()->getStopIndex() + 1,
+                                                                std::make_pair("", " ON TRUE ")));
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:1018]: Could you add a test case like `A outer apply B outer apply C` if there is no such test?

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 3]
```diff
@@ -986,6 +986,34 @@ class tsqlCommonMutator : public TSqlParserBaseListener
...
+            }
+            else
+            {
+                return; // TODO implement error message
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: Is there a reason we cannot provide an error message now?

Comment (Sairakan) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp]: TBH I don't think this is needed - the parser will raise an error anyways due to incorrect syntax. I don't know if we care whether the error message exactly matches SQL Server's error message

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 2 above)
Comment (Sairakan) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:1018]: Added

File: test/JDBC/input/BABEL-2805.sql (hunk 1 above)
Comment (Sairakan) [reply test/JDBC/input/BABEL-2805.sql]: Added

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 4]
```diff
@@ -992,26 +992,42 @@ class tsqlCommonMutator : public TSqlParserBaseListener
...
+                if (lateral)
+                {
+                    rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
+                                                                    std::make_pair(ctx->APPLY()->getText(), "JOIN LATERAL")));
+                }
+                else
+                {
+                    rewritten_query_fragment.emplace(std::make_pair(ctx->APPLY()->getSymbol()->getStartIndex(),
+                                                                    std::make_pair(ctx->APPLY()->getText(), "JOIN")));
+                }
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp]: Why not using `lateral ? "JOIN LATERAL" : "JOIN"`?

File: test/JDBC/expected/BABEL-328.out [hunk 5]
```diff
@@ -165,15 +165,63 @@ GO
...
+2#!#2#!#4#!#4#!#5#!#5
+2#!#2#!#3#!#3#!#6#!#6
+2#!#2#!#4#!#4#!#6#!#6
+~~END~~
+
+
+drop table t1
+go
+
+drop table t2
```
Comment (2jungkook) [inline test/JDBC/expected/BABEL-328.out:225]: 1. Please follow our object name conventions, e.g., `babel_328_vu_xxx`.
2. Our test strategy is to add upgrade tests unless it is unnecessary or impossible. So, please consider to move this to upgrade tests using views.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 4 above)
Comment (Sairakan) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp]: good point, I'll use that

File: test/JDBC/expected/BABEL-328.out (hunk 5 above)
Comment (Sairakan) [reply test/JDBC/expected/BABEL-328.out:225]: Moved all tests to upgrade tests

**Review by 2jungkook — APPROVED** [review]

**Pull request merged**